packages/spawner/src/contractHost.js

```diff
@@ -47,127 +47,76 @@ function makeContractHost(E, evaluate, additionalEndowments = {}) {
-    );
-    return fn;
-  }
+  // note: support for check functions was removed during warner's
+  // new-SES-ification, there were no (enabled) tests to exercise them. The
+  // comments that talk about special treatment of 'check' functions should
+  // be treated with suspicion.
 
-  /**
-   * Build an object containing functions with names starting with 'check' from
-   * strings in the input contract.
-   */
-  function extractCheckFunctions(contractSrcs) {
-    const installation = {};
-    for (const fname of Object.getOwnPropertyNames(contractSrcs)) {
-      if (typeof fname === 'string' && fname.startsWith('check')) {
-        const fn = evaluateStringToFn(contractSrcs[fname]);
-        installation[fname] = (...args) => fn(installation, ...args);
-      }
-    }
-    return installation;
-  }
+  // TODO: We'd like to have fail-stop semantics, which means we associate a
+  // meter with a spawn and not with an installation, and failed spawns die
+  // forever. Check functions, on the other hand, should be billed to the
+  // installation, which may die forever.
 
   /** The contract host is designed to have a long-lived credible identity. */
   const contractHost = harden({
     getInviteIssuer() {
       return inviteIssuer;
     },
-    // contractSrcs is a record containing source code for the functions
-    // comprising a contract. `spawn` evaluates the `start` function
-    // (parameterized by `terms` and `inviteMaker`) to start the contract, and
-    // returns whatever the contract returns. The contract can also have any
-    // number of functions with names beginning 'check', each of which can be
-    // used by clients to help validate that they have terms that match the
-    // contract.
-    install(contractSrcs, moduleFormat = 'object') {
-      let installation;
-      if (moduleFormat === 'object') {
-        installation = extractCheckFunctions(contractSrcs);
-      } else if (
-        moduleFormat === 'getExport' ||
-        moduleFormat === 'nestedEvaluate'
+    // contractBundle is a record containing source code for the functions
+    // comprising a contract, as created by bundle-source. `spawn` evaluates
+    // the `start` function (parameterized by `terms` and `inviteMaker`) to
+    // start the contract, and returns whatever the contract returns. The
+    // contract can also have any number of functions with names beginning
+    // 'check', each of which can be used by clients to help validate that
+    // they have terms that match the contract.
+
+    // TODO: we have 2 or 3 dapps (in separate repos) which do { source,
+    // moduleFormat } = bundleSource(..), then E(spawner).install(source,
+    // moduleFormat). Those will get the default
+    // moduleFormat="nestedEvaluate". We need to support those callers, even
+    // though our new preferred API is just install(bundle). We also look for
+    // getExport because that's easier to create in the unit tests. TODO once
```

Comment:
Please start "TODO" on a new line. Some tools treat comments that start with TODO special.